src/openvic-simulation/economy/production/ResourceGatheringOperation.cpp

```diff
@@ -293,32 +293,27 @@ void ResourceGatheringOperation::pay_employees(
 	ProvinceInstance& location,
 	const fixed_point_t revenue,
 	const Pop::pop_size_t total_worker_count_in_province,
-	std::vector<Pop*>& owner_pops_cache,
+	std::vector<Pop*>* const owner_pops_cache,
 	const Pop::pop_size_t total_owner_count_in_state_cache
 ) {
 	total_owner_income_cache = 0;
 	total_employee_income_cache = 0;
-	if (production_type_nullable == nullptr || revenue <= 0 || total_worker_count_in_province <= 0) {
+	if (revenue <= 0 || total_worker_count_in_province <= 0) {
 		if (revenue < 0) { Logger::error("Negative revenue for province ", location.get_identifier()); }
 		if (total_worker_count_in_province < 0) { Logger::error("Negative total worker count for province ", location.get_identifier()); }
 		return;
 	}
-	
-	ProductionType const& production_type = *production_type_nullable;
 
 	fixed_point_t revenue_left = revenue;
-	if (total_owner_count_in_state_cache > 0) {
-		Job const& owner_job = production_type.get_owner().value();
-		PopType const* owner_job_pop_type_nullable = owner_job.get_pop_type();
-		
+	if (total_owner_count_in_state_cache > 0) {		
 		fixed_point_t owner_share = (fixed_point_t::_2() * total_owner_count_in_state_cache / total_worker_count_in_province);
 		constexpr fixed_point_t upper_limit = fixed_point_t::_0_50();
 		if (owner_share > upper_limit) {
 			owner_share = upper_limit;
 		}
 
-		for(Pop* owner_pop_nullable : owner_pops_cache) {
-			Pop& owner_pop = *owner_pop_nullable;
+		for(Pop* owner_pop_ptr : *owner_pops_cache) {
```

Comment:
Are we sure `owner_pops_cache` can't be null here? If it can then we need to check that earlier in the function, if it can't then the function should take a reference rather than a pointer.

Reply:
Good point. I messed up here. If there is no owner, we should exit early. Empty provinces just don't tick. I ensured this by looping through the states but the RGO tick can simply check for owner.

Reply:
Yes, because `total_owner_count_in_state_cache > 0` and that can only happen when there is an owner pop type. If there is an owner pop type, the owner_pops_cache is filled.